Tests/RestKitTests/CodableExtensionsTests.swift

```diff
@@ -0,0 +1,482 @@
+        XCTAssertEqual(json, "{\"key\":[null,true,1,0.5,\"this is a string\"]}")
+    }
+    
+    func testEncodeTopLevelArray() {
+        let array: [Any] = [NSNull(), true, 1, 0.5, "this is a string"]
+        let data = try! JSONEncoder().encode(array)
+        let json = String(data: data, encoding: .utf8)
+        XCTAssertEqual(json, "[null,true,1,0.5,\"this is a string\"]")
+    }
+    
+    func testEncodeNestedArrays() {
+        let array: [Any] = [[1, 2, 3], [4, 5, 6]]
+        let data = try! JSONEncoder().encode(array)
+        let json = String(data: data, encoding: .utf8)
+        XCTAssertEqual(json, "[[1,2,3],[4,5,6]]")
+    }
+    
+    func testEncodeArrayOfObjects() {
+        let array: [Any] = [["x": 1], ["y": 2], ["z": 3]]
+        let data = try! JSONEncoder().encode(array)
+        let json = String(data: data, encoding: .utf8)
+        XCTAssertEqual(json, "[{\"x\":1},{\"y\":2},{\"z\":3}]")
+    }
+
+    func testEncodeObject() {
+        let object: [String: Any] = ["key": ["null": NSNull(), "bool": true, "int": 1, "double": 0.5, "string": "this is a string"]]
+        let data = try! JSONEncoder().encode(object)
+        let json = String(data: data, encoding: .utf8)
+        XCTAssertEqual(json, "{\"key\":{\"bool\":true,\"double\":0.5,\"string\":\"this is a string\",\"int\":1,\"null\":null}}")
+    }
+    
+    func testEncodeEmptyObject() {
+        let object = [String: Any]()
+        let data = try! JSONEncoder().encode(object)
+        let json = String(data: data, encoding: .utf8)
+        XCTAssertEqual(json, "{}")
+    }
+
+    func testEncodeNested() {
+        let object: [String: Any] = ["key": ["array": [1, 2, 3], "object": ["x": 1, "y": 2, "z": 3]]]
+        let data = try! JSONEncoder().encode(object)
+        let json = String(data: data, encoding: .utf8)
+        XCTAssertEqual(json, "{\"key\":{\"array\":[1,2,3],\"object\":{\"y\":2,\"x\":1,\"z\":3}}}")
+    }
+    
+    func testEncodeDeeplyNested() {
+        let object: [String: Any] = ["key1": ["key2": ["key3": ["key4": [1, 2, 3]]]]]
+        let data = try! JSONEncoder().encode(object)
+        let json = String(data: data, encoding: .utf8)
+        XCTAssertEqual(json, "{\"key1\":{\"key2\":{\"key3\":{\"key4\":[1,2,3]}}}}")
+    }
+    
+    func testEncodeMetadataModel() {
+        let metadata: [String: Any] = ["null": NSNull(), "bool": true, "int": 1, "double": 0.5, "string": "this is a string"]
+        let model = TestModel(name: "name", metadata: metadata, additionalProperties: [:])
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let data = try! encoder.encode(model)
+        let json = String(data: data, encoding: .utf8)
+        let expected = """
+            {
+              \"name\" : \"name\",
+              \"metadata\" : {
+                \"bool\" : true,
+                \"double\" : 0.5,
+                \"string\" : \"this is a string\",
+                \"int\" : 1,
```

Comment:
Metadata could also contain arrays and objects.  We should include an array and object property in the metadata of this test to verify handling of those types.

Reply:
Great idea! Added an array and object to the updated tests.